Resources/Prototypes/Entities/Structures/Doors/Airlocks/assembly.yml

```diff
@@ -36,6 +36,30 @@
     sprite: Structures/Doors/Airlocks/Glass/cargo.rsi
     state: "assembly"
 
+#Clockwork
+- type: entity
+  id: ClockworkAirlockAssembly
+  name: airlock assembly
+  parent: AirlockAssembly
+  components:
+  - type: Sprite
+    sprite: Structures/Doors/Airlocks/Standard/clockwork_pinion.rsi
+    state: "assembly"
+  - type: Construction
+    graph: ClockworkAirlock
+    node: assembly
+  placement:
+    mode: SnapgridCenter
+
+- type: entity
+  parent: ClockworkAirlockAssembly
+  id: ClockworkAirlockAssemblyGlass
```

Comment:
it's identical to the way non-clockwork glass airlocks are constructed, where a glass airlock is made by inserting two glass after the door electronics instead of screwing.